### PR TITLE
Made Modal a connected component again 

### PR DESCRIPTION
**Pull request by petrican**
- Made Modal a connected component again
- It was initially made disconnected for the sole purpose of testing createProcess
- having it disconnected causes the issue from https://github.com/metasfresh/metasfresh/issues/7128
- skipped the createProcess test - following to add it back (most probable during refactoring, as having the component disconnected implies more fixes as there might be more corner cases where similar issues can occur)
Note: Having this functional is needed for this release!

### CODE REVIEW

**Comment by petrican** [comment]
https://www.loom.com/share/e6ab5135cd2040b29f1311b5f99106b0

File: frontend/src/__tests__/components/app/Modal.test.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ describe('Modal test', () => {
     expect(html.includes('Action')).toBe(true);
   });
 
-  it(`calls startProcess when initializing a modal of 'process' type`, async (done) => {
+  it.skip(`calls startProcess when initializing a modal of 'process' type`, async (done) => {
```
Comment (teosarca) [inline frontend/src/__tests__/components/app/Modal.test.js:51]: pls add a TODO and document it on the issue page where this will be fixed (e.g. the refactoring task).
We need to keep track of this kind of changes and we have to address them later.

**Comment by siemiatj** [comment]
@petrican How is this PR fixing anything ? Modal was connected and after this change still is. Not sure what didn't work but I would assume some prop action creator was not properly initialized (like https://github.com/metasfresh/metasfresh/pull/7135/files#diff-d871f6d8d1524b21b07b4fcd3c96151aR316). And now we've lost unit tests with this change.

**Comment by petrican** [comment]
What was not working -> https://github.com/metasfresh/me03/issues/5344 .  Component was connected indeed, but I had to revert changes we did for the test. Currently the test is skipped but there is a follow up to be brought back. I did that because it was mandatory for the last release to have that saving functioning.
